assayist/client/query.py

```diff
@@ -18,6 +18,51 @@ def set_connection(neo4j_url):
     neomodel.db.set_connection(neo4j_url)
 
 
+def get_container_by_component(name, type_, version):
+    """Query for builds by component name and version.
+
+    Finds builds that produced container images which include or embed a component with the
+    specified name, type, and version.
+
+    :param str name: the canonical name of the component
+    :param str type_: the canonical type of the component
+    :param str version: the canonical version of the component
+    :return: list of builds IDs
+    :rtype: list
+    """
+    query = """
+        // Find a component and a sourcelocation with the specified name, type and version. These
+        // can be emdedded in other sourcelocation that are used as a source for a build or directly
+        // used for a build.
+        MATCH (c:Component {{canonical_name: '{name}', canonical_type: '{type}'}}) <-[:SOURCE_FOR]-
+              (version_sl:SourceLocation {{canonical_version: '{version}'}})
+              <-[:UPSTREAM|EMBEDS*0..]- (target_sl:SourceLocation)
```

Comment:
So if you pass in an internal only canonical version then all the source locations that embed it will be found. If you pass in an upstream only canonical version, then the embeds will have no effect because the upstream source locations will not have embeds. Is this desired behavior?

Perhaps what you want to do is:
```
MATCH (c:Component {{canonical_name: '{name}', canonical_type: '{type}'}}) <-[:SOURCE_FOR]-
    (version_sl:SourceLocation {{canonical_version: '{version}'}})
    <-[:UPSTREAM*0..]-(:SourceLocation)<-[:EMBEDS*0..]-(target_sl:SourceLocation)
```